sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -1110,13 +1103,13 @@ object SparkSession extends Logging {
   private[sql] def getOrCloneSessionWithConfigsOff(
       session: SparkSession,
       configurations: Seq[ConfigEntry[Boolean]]): SparkSession = {
-    val configsEnabled = configurations.filter(session.conf.get[Boolean])
+    val configsEnabled = configurations.filter(session.sessionState.conf.getConf[Boolean])
```

Comment:
Although this is not a breaking change, do we still need this change inevitably for some reasons?

Reply:
If then, the developers seem to be annoyed by this change.

Reply:
In this PR I use the abstract class in sql/api the main interface. I did not really wanted to add support for this because it is not user facing, and because it requires me to move even more stuff into sql/api.

Reply:
Which developers? Spark developers?